src/index/updater.rs

```diff
@@ -96,27 +98,46 @@ impl Updater {
       Some(progress_bar)
     };
 
-    let rx = Self::fetch_blocks_from(index, self.height, self.index_sats)?;
-
-    let (mut outpoint_sender, mut value_receiver) = Self::spawn_fetcher(index)?;
+    let rx = Self::fetch_blocks_from(self.index, self.height, self.index_sats)?;
 
     let mut uncommitted = 0;
-    let mut value_cache = HashMap::new();
     while let Ok(block) = rx.recv() {
-      self.index_block(
-        index,
-        &mut outpoint_sender,
-        &mut value_receiver,
-        &mut wtx,
-        block,
-        &mut value_cache,
-      )?;
+      if self.index.client.get_block_count()? - self.index.block_height()?.unwrap_or(Height(0)).n()
+        < 50
+      {
+        log::info!("not in initial sync; starting to create savepoints");
+        self.initial_sync = false;
+
+        if self.savepoints.len() >= 6 {
+          dbg!("here");
+          wtx.delete_persistent_savepoint(self.savepoints.pop_front().unwrap())?;
+        }
+
+        let savepoints: Vec<u64> = wtx.list_persistent_savepoints()?.into_iter().collect();
+        dbg!(&savepoints);
+        // self.savepoints.push_back(wtx.ephemeral_savepoint()?);
```

Comment:
I think you want to create a persistent save_point here. Also, you'll need to commit wtx and reinitialise it directly after.

Reply:
Since you have to commit the txn and reinitialise it, it might make more sense for this logic to live directly after a `self.commit(wtx)` call or maybe even inside the commit function.

Reply:
In fact, I think wtx has to be clean (no unflushed writes) before creating the savepoint, and you have to commit straight after it as it would make the wtx dirty.